FirebaseAdmin/FirebaseAdmin/Auth/FirebaseUserManager.cs

```diff
@@ -0,0 +1,167 @@
+                { "INVALID_CLAIMS", "invalid-claims" },
+                { "INVALID_EMAIL", "invalid-email" },
+                { "INVALID_PAGE_SELECTION", "invalid-page-token" },
+                { "INVALID_PHONE_NUMBER", "invalid-phone-number" },
+                { "PHONE_NUMBER_EXISTS", "phone-number-already-exists" },
+                { "PROJECT_NOT_FOUND", "project-not-found" },
+                { "USER_NOT_FOUND", "user-not-found" },
+                { "WEAK_PASSWORD", "invalid-password" },
+                { "UNAUTHORIZED_DOMAIN", "unauthorized-continue-uri" },
+                { "INVALID_DYNAMIC_LINK_DOMAIN", "invalid-dynamic-link-domain" }
+            });
+
+        private const string ID_TOOLKIT_URL = "https://identitytoolkit.googleapis.com/v1/projects/{0}";
+
+        private readonly ConfigurableHttpClient _httpClient;
+        private readonly string _baseUrl;
+
+        private FirebaseUserManager(FirebaseUserManagerArgs args)
+        {
+            _httpClient = args.ClientFactory.CreateAuthorizedHttpClient(args.Credential);
+            _baseUrl = string.Format(ID_TOOLKIT_URL, args.ProjectId);
+        }
+
+        /// <summary>
+        /// Update an existing user.
+        /// </summary>
+        /// <exception cref="FirebaseAuthException">If the server responds that cannot update the user.</exception>
+        /// <param name="user">The user which we want to update.</param>
+        public async Task UpdateUser(UserRecord user)
+        {
+            var updatePath = "/accounts:update";
+            var resopnse = await PostAsync(updatePath, user);
+
+            var userResponse = resopnse.ToObject<UserRecord>();
+            if (userResponse == null || userResponse.Uid != user.Uid)
+            {
+                throw new FirebaseAuthException(INTERNAL_ERROR, $"Failed to update user: {user.Uid}");
+            }
+        }
+
+        private async Task<JObject> PostAsync(string path, UserRecord user)
+        {
+            var requestUri = $"{_baseUrl}{path}";
+            HttpResponseMessage response = null;
+            try
+            {
+                response = await _httpClient.PostJsonAsync(requestUri, user, default);
+
+                if (response.IsSuccessStatusCode)
+                {
+                    return JObject.Parse(await response.Content.ReadAsStringAsync());
+                }
+                else
+                {
+                    await HandleHttpError(response);
+                }
+            }
+            catch (Exception)
+            {
+                throw new FirebaseAuthException(INTERNAL_ERROR, "Error while calling user management backend service");
+            }
+            finally
+            {
+                if (response != null)
+                {
+                    response.Dispose();
+                }
+            }
+
+            return null;
+        }
+
+        private async Task HandleHttpError(HttpResponseMessage response)
```

Comment:
We haven't quite settled on the error handling strategy for .NET yet. For now we should just throw an FirebaseException here.

Reply:
Does that mean that we should just `throw new FirebaseException(code);` or to not to deal with server error codes at all for now?